### ISSUE TITLE
precision and recall values kept unchanged for some training steps.

### ISSUE THREAD

**Issue by hhzrz** [tensorflow/tensorflow#19138]
### System information
- Have I written custom code: No, i just use a canned estimator DNNClassifier.
- OS Platform and Distribution: macOS High Sierra
- TensorFlow installed from: pip
- TensorFlow version: 1.8.0
- Bazel version: N/A
- CUDA/cuDNN version: N/A
- GPU model and memory: N/A
- Exact command to reproduce: N/A

### Describe the problem
I'm using google ml-engine to train a model for ctr-prediction(the training dataset is Imbalanced).
Following is the training log screenshot from ml-engine.
![pr-auc](https://user-images.githubusercontent.com/17157194/39740932-52b01a7c-52ca-11e8-8952-820424e38357.jpg)
The problem is: the **precision** and **recall** values kept unchanged for some training steps as i annotated in the picture(auctually these two values kept unchanged for 2 epochs in this picture). I'm pretty sure this is a bug, but i don't know where is wrong.
Does any one can check the calculation of these two metrics(precision, recall) for me?

**Comment by tensorflowbutler** [comment]
Nagging Assignee @rohan100jain: It has been 16 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @rohan100jain: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @rohan100jain: It has been 48 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @rohan100jain: It has been 63 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @rohan100jain: It has been 78 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by rohan100jain** [comment]
Can you please provide details as to what commands you ran so that we can reproduce and check?

**Comment by hhzrz** [comment]
gcloud ml-engine local train \
    --package-path trainer \
    --module-name trainer.task \
    --job-dir ${job_dir} \
    --distributed \
    -- \
    --train_files ${train_files} \
    --eval_files ${eval_files} \
    --hidden_units ${hidden_units} \
    --cats_path ${cats_path} \
    --job_dir ${job_dir} \
    --train_batch_size ${train_batch_size} \
    --eval_batch_size ${eval_batch_size} \
    --train_set_size ${train_set_size} \
    --num_epochs ${num_epochs} \`

**Comment by tensorflowbutler** [comment]
Nagging Assignee @rohan100jain: It has been 18 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @rohan100jain: It has been 35 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @rohan100jain: It has been 50 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @rohan100jain: It has been 65 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @rohan100jain: It has been 80 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @rohan100jain: It has been 95 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @rohan100jain: It has been 110 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.
